### ISSUE TITLE
On building, the site needs to validate all URL's used on the pages to ensure they are valid

### ISSUE THREAD

**Issue by SimonDarksideJ** [MonoGame/monogame.github.io#80]
### Description

SUch URL's needing validating are:

* Links to external content
* Links to other pages on the MonoGame docs site
* Links to images

The testing only needs to check if the link is valid (non 404) rather than its contents.

**Comment by AristurtleDev** [comment]
Currently in the process of creating this if ya'll want to assign the issue to me

**Comment by SimonDarksideJ** [comment]
Slightly extended request on checking links.
If possible, can the console dump use different colours for API link errors and Documentation link errors?

Or have two output files for each? @AristurtleDev

**Comment by AristurtleDev** [comment]
> Slightly extended request on checking links. If possible, can the console dump use different colours for API link errors and Documentation link errors?
> 
> Or have two output files for each? @AristurtleDev

Yea, that's not an issue. Just to clarify, you want it separated based on the document type, not where the external link is going correct?

For instance, if an `/article` document links to an `/api` document, but the link is incorrect, then this should be output as part of the **article** grouping and not the **api** grouping.

**Comment by SimonDarksideJ** [comment]
Currently, the output for each issue is as follows:

* Detail on each specific issue
* A collective report of the affected lines

But with the screen FULL of these, it is hard to follow through one by one.

But colour coding which reports are /API and which are /Documentation will help, since different teams are viewing each area.
SO just a distinction between API and documentation should be sufficient for now.
